model/device.go

```diff
@@ -78,6 +133,100 @@ func (da DeviceAttribute) Validate() error {
 	)
 }
 
+func allowedType(v reflect.Value) bool {
+	if v.Type().Kind() == reflect.Int {
+		return true
+	}
+	if v.Type().Kind() == reflect.String {
+		return true
+	}
+	if v.Type().Kind() == reflect.Float64 {
+		return true
+	}
+	return false
+}
+
+func reflectValuesEqual(rVal1 reflect.Value, rVal2 reflect.Value) bool {
+	if rVal1.Len() != rVal2.Len() {
+		return false
+	}
+	for i := 0; i < rVal1.Len(); i++ {
+		if rVal1.Index(i).Kind() != rVal2.Index(i).Kind() {
+			return false
+		}
+		if !allowedType(rVal1.Index(i)) || !allowedType(rVal2.Index(i)) {
+			if !reflect.DeepEqual(rVal1.Index(i), rVal2.Index(i)) {
+				return false
+			}
+		} else {
+			if rVal2.Index(i).Kind() == reflect.Int {
+				value1 := rVal1.Index(i).Interface().(int)
+				value2 := rVal2.Index(i).Interface().(int)
+				if value1 != value2 {
+					return false
+				}
+			}
+			if rVal2.Index(i).Kind() == reflect.String {
+				value1 := rVal1.Index(i).Interface().(string)
+				value2 := rVal2.Index(i).Interface().(string)
+				if value1 != value2 {
+					return false
+				}
+			}
+			if rVal2.Index(i).Kind() == reflect.Float64 {
+				floatComparePrecision := "%.8f"
+				floatValue1 := rVal1.Index(i).Interface().(float64)
+				floatValue2 := rVal2.Index(i).Interface().(float64)
+				value1 := fmt.Sprintf(floatComparePrecision, floatValue1)
+				value2 := fmt.Sprintf(floatComparePrecision, floatValue2)
```

Comment:
I still don't get why you're going through this string conversion gymnastic. We don't do any floating point arithmetic so I don't think it's necessary to consider any potential floating point rounding errors on the client.

Reply:
lets say it is just in case.